spec/controllers/api/v1/products_controller_spec.rb

```diff
@@ -404,4 +404,44 @@ def make_request
       end
     end
   end
+
+  describe 'POST #transfer' do
+    def target_user
+      @target_user ||= create(:user)
+    end
+
+    def make_request(product, target_email)
+      authenticate_user(user)
+      post :transfer, params: { id: product, email: target_email }
+    end
+
+    context 'when is successful' do
+      it 'returns the product of the target user' do
+        make_request(user.products.first.id, target_user.email)
+        expect(JSON.parse(response.body)['user_id']).to eql(target_user.products.first.user_id)
+      end
+    end
```

Comment:
What about the http code here? are we using a serializer here? maybe we need to check the response to check it, what do you think?